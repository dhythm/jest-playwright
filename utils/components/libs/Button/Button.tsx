import { Button as MtButton } from '@mantine/core';
import { ComponentProps } from 'react';
import React from 'react';

type Props = Omit<ComponentProps<typeof MtButton>, 'sx'>;
export const Button = (props: Props) => {
  return (
    <MtButton
      sx={(theme) => ({
        backgroundColor: '#426782',
        color: '#fff',
        '&:hover': {
          backgroundColor: theme.fn.darken('#4267B2', 0.1),
        },
      })}
      {...props}
    />
  );
};
