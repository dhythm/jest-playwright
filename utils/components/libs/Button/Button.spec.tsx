import { MantineProvider } from '@mantine/core';
import { render } from '@testing-library/react';
import { Button } from './Button';

test('render DOM', () => {
  const view = render(
    <MantineProvider withGlobalStyles withNormalizeCSS>
      <Button>Button</Button>
    </MantineProvider>
  );
  expect(view.container).toMatchSnapshot();
});
