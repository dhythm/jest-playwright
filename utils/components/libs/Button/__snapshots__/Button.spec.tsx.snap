// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render DOM 1`] = `
<div>
  <button
    class="mantine-UnstyledButton-root mantine-Button-root mantine-1m7qy9i"
    data-button="true"
    type="button"
  >
    <div
      class="mantine-3xbgk5 mantine-Button-inner"
    >
      <span
        class="mantine-qo1k2 mantine-Button-label"
      >
        Button
      </span>
    </div>
  </button>
</div>
`;
